### DISCUSSION TITLE
Relation to specific row and to many rows in same table?

### DISCUSSION THREAD

**Discussion by glowstab** [prisma/prisma#20662] [Q&A]
### **Similar questions have been asked, maybe don't waste your time on this one. I'm mostly confused about performance and theory.**

I have a project that includes a few hundred items (the number of items almost never changes). Each `item` has potentially hundreds of historical prices.
```schema
model Item {
   id   Int   @unique
   ...
   PriceHistory Price[]
}

model Price {
   item   Item   @relation(fields: [itemId], references: [id])
   itemId   Int
   value   Int
   updatedAt   Date
}
```
Currently I am achieving this with 
```
await prisma.item.findMany({
    where: { name },
    include: {
      PriceHistory: {
        orderBy: {
          updatedAt: "desc",
        },
        take: 1,
      },
    },
  });
```
But based on the raw query this generates, it needs to get the item itself, then sort the entire `price` table, and then select the first one with the correct `name`. 

I would like to speed this process up by keeping a reference to the most recent `price` object inside the `item` itself. This application has **many** more reads than writes and the vast majority of the time we are interested in the most recent price object, but we want to gather historical data and potentially use it in the future.

I imagine something like this:
```schema
model Item {
   id   Int   @unique
   ...
   RecentPrice Price
   PriceHistory Price[]
}

```

1. Is there a way to have a `one-to-one` and `one-to-many` relationship with the same two models? (My naive approach failed because one-to-one requires a `@unique` and one-to-many cannot have it)
2. Is this even a problem in the first place? I am newish to SQL in general so I am not sure, but intuitively sorting thousands of prices to find the one I already know I want seems bad.
3. Would storing the `id` of the recent `price` in the `item` model, and then finding the correct `price` be better?

I'm sure this is basic stuff for some but it is certainly causing me confusion. Thanks so much for the help.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by janpio** [comment]
> But based on the raw query this generates, it needs to get the item itself, then sort the entire `price` table, and then select the first one with the correct `name`.

What version of Prisma are you using?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
